packages/rum/src/viewCollection.ts

```diff
@@ -148,8 +150,17 @@ function newView(
     })
   }
 
+  function updateLoadingTime(newLoadingTime: number) {
+    if (loadingTime === undefined || newLoadingTime > loadingTime) {
+      loadingTime = newLoadingTime
+    }
+  }
+
   function updateMeasures(newMeasures: Partial<ViewMeasures>) {
     measures = { ...measures, ...newMeasures }
+    if (measures.loadEventEnd !== undefined) {
+      updateLoadingTime(nsToMs(measures.loadEventEnd))
+    }
```

Comment:
Several things are bothering me with this strategy:

- `trackLoadingTime` was responsible to handle all the loading time logic and by looking at it we could easily understand the functional rule that the loading time should take event load into account on initial view. Now, it depends on the fact that measures are updated for the initial view which is not as clear.
- updateMeasures is then not only updating measures but also loading time
- having to check the loading time value before updating it and doing the `nsToMs` conversion let me believe that there is a design issue.

What about having a strategy allowing us to keep a design similar to the previous one?

- `trackLoadingTime` could expose `onTimingValue` instead of having `trackLoadEventLoadingTime` and it would be called when an update if the load event occurs

- splitting `updateMeasures` into `updateTimings` and `updateCounters` would allow to better follow what is relative to timings and what is relative to counters. It would also simplify the v2 migration.

- what about doing the `nsToMs` conversion later to avoid the opposite conversion?

Reply:
I implemented a new version of `trackLoadingTime` which fixes the loadingTime computation (= only set it once). It does not embed the `trackActivityLoadingTime` call, because calling `stop` should be final, and `trackLoadingTime` should not be stoped at the view end since it is still useful after. LMKWYT